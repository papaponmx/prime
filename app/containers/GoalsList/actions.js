/*
 *
 * GoalsList actions
 *
 */

import {
  DELETE_GOAL,
  DELETE_GOAL_ERROR,
  DELETE_GOAL_SUCCESS,
  FETCH_GOALS,
  FETCH_GOALS_ERROR,
  FETCH_GOALS_SUCCESS,
  ADD_GOAL,
  ADD_GOAL_ERROR,
  ADD_GOAL_SUCCESS,
  MARK_GOAL_AS_DONE,
} from './constants';

export function fetchGoals() {
  return {
    type: FETCH_GOALS,
  };
}

export const fetchGoalsError = payload => ({
  type: FETCH_GOALS_ERROR,
  payload,
});

export const fetchGoalsSuccess = goals => ({
  type: FETCH_GOALS_SUCCESS,
  payload: goals,
});

export function deleteGoal(goalId) {
  return {
    type: DELETE_GOAL,
    payload: goalId,
  };
}

export const deleteGoalError = payload => ({
  type: DELETE_GOAL_ERROR,
  payload,
});

export const deleteGoalSuccess = goals => ({
  type: DELETE_GOAL_SUCCESS,
  payload: goals,
});

export const addGoal = values => ({
  type: ADD_GOAL,
  payload: values,
});

export const addGoalError = values => ({
  type: ADD_GOAL_ERROR,
  payload: values,
});

export const addGoalSuccess = values => ({
  type: ADD_GOAL_SUCCESS,
  payload: values,
});

export const markGoalAsDone = goalId => ({
  type: MARK_GOAL_AS_DONE,
  payload: goalId,
});
