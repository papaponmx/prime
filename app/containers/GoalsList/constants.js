/*
 *
 * GoalsList constants
 *
 */

export const ADD_GOAL = 'app/GoalsList/ADD_GOAL';
export const ADD_GOAL_ERROR = 'app/GoalsList/ADD_GOAL_ERROR';
export const ADD_GOAL_SUCCESS = 'app/GoalsList/ADD_GOAL_SUCCESS';

export const DELETE_GOAL = 'app/GoalsList/DELETE_GOAL';
export const DELETE_GOAL_ERROR = 'app/GoalsList/DELETE_GOAL_ERROR';
export const DELETE_GOAL_SUCCESS = 'app/GoalsList/DELETE_GOAL_SUCCESS';

export const FETCH_GOALS = 'app/GoalsList/FETCH_GOALS';
export const FETCH_GOALS_ERROR = 'app/GoalsList/FETCH_GOALS_ERROR';
export const FETCH_GOALS_SUCCESS = 'app/GoalsList/FETCH_GOALS_SUCCESS';
