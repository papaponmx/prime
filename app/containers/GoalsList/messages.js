/*
 * GoalsList Messages
 *
 * This contains all the text for the GoalsList component.
 */

import { defineMessages } from 'react-intl';

export default defineMessages({
  header: {
    id: 'app.containers.GoalsList.header',
    defaultMessage: 'This is GoalsList container !',
  },
});
