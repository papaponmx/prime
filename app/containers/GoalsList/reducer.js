/*
 *
 * GoalsList reducer
 *
 */

import {
  FETCH_GOALS,
  FETCH_GOALS_ERROR,
  FETCH_GOALS_SUCCESS,
} from './constants';

export const initialState = {
  isLoading: false,
  isLoaded: false,
  list: [],
  error: null,
};

function goalsListReducer(state = initialState, { type, payload }) {
  switch (type) {
    case FETCH_GOALS:
      return {
        ...state,
        isLoading: true,
      };
    case FETCH_GOALS_ERROR:
      return {
        ...state,
        isLoading: false,
        error: payload,
      };
    case FETCH_GOALS_SUCCESS:
      return {
        ...state,
        isLoading: false,
        isLoaded: true,
        list: payload,
      };
    default:
      return state;
  }
}

export default goalsListReducer;
