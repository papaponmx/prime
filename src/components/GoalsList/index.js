import { connect } from 'react-redux'
import React, { Component } from 'react'
import ListItem from '@material-ui/core/ListItem'
import List from '@material-ui/core/List';

class GoalsList extends Component {
  // TODO: Fisrt Optimize this component
  // IDEA: Add role badge to each list item
  renderList () {
    const {goals} = this.props
    return goals.list.map((goal) => {
      const { goalName } = goal
      return (
        <List>
          <ListItem key={goalName}>
            {goalName}
          </ListItem>
        </List>
      )
    })
  }

  render () {
    return (
      <div>
        {(this.props.goals.list)
           ? this.renderList()
           : `Please start by adding some goals`}
      </div>
    )
  }
}

const mapStateToProps = state => ({
  goals: state.goals.toJS()
})

export default connect(mapStateToProps, null)(GoalsList)
