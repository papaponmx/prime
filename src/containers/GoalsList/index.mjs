import React, { Component } from 'react'
import { connect } from 'react-redux'
import { fetchGoals } from '../../actions'

export class GoalsList extends Component {
  componentDidMount () {
    if (!this.props.goals.fetched) {
      this.props.fetchGoals()
    }
  }

  render () {
    const {       goals, fetched    } =  this.props;

    return (
      <div>
        <h3>These are your Goals</h3>
        <ul>
          {fetched && goals.map( ({ id, name}) => (
            <li key={id}>
              {name}
            </li>
          ))}
        </ul>
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  goals: state.goals.list,
  fetched: state.goals.fetched,
})

const mapDispatchToProps = (dispatch) => ({
  fetchGoals: () => dispatch(fetchGoals()),
})

export default connect(mapStateToProps, mapDispatchToProps)(GoalsList)
