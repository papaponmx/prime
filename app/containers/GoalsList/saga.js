// import firebase from 'firebase/app';
import { takeLatest, put, select } from 'redux-saga/effects';
import { FETCH_GOALS } from './constants';
import { fetchGoalsSuccess, fetchGoalsError } from './actions';
import { database as db } from '../../configureStore';
import { selectUserUid } from '../LoginPage/selectors';

export function* fetchGoals() {
  let list;
  const uid = yield select(selectUserUid);
  yield db.settings({ timestampsInSnapshots: true });

  try {
    yield db
      .collection('users')
      .doc(uid)
      .collection('goals')
      .get()
      .then(res => {
        list = res.docs.map(goal => goal.data());
        return list;
      });
  } catch (error) {
    yield put(fetchGoalsError(error));
  }
  yield put(fetchGoalsSuccess(list));
}

// Individual exports for testing
export default function* defaultSaga() {
  yield takeLatest(FETCH_GOALS, fetchGoals);

  // See example in containers/HomePage/saga.js
}
