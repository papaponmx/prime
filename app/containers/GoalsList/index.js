/**
 *
 * GoalsList
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { FormattedMessage } from 'react-intl';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';
import makeSelectGoalsList from './selectors';
import reducer from './reducer';
import saga from './saga';
import messages from './messages';
import { fetchGoals } from './actions';
import List from '../../components/List';

/* eslint-disable react/prefer-stateless-function */
export class GoalsList extends React.Component {
  componentDidMount() {
    // if (!this.props.goals.isLoaded && !this.props.goals.isLoading) {
    //   this.props.fetchGoals();
    // }
  }

  render() {
    const { list, isLoaded } = this.props.goals;
    return (
      <div>
        <FormattedMessage {...messages.header} />
        {isLoaded && <List list={list} listType="goals" />}
      </div>
    );
  }
}

GoalsList.propTypes = {
  // fetchGoals: PropTypes.func,
  goals: PropTypes.object,
};

const mapStateToProps = createStructuredSelector({
  goals: makeSelectGoalsList(),
});

const mapDispatchToProps = dispatch => ({
  fetchGoals: dispatch(fetchGoals()),
});

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

const withReducer = injectReducer({ key: 'goalsList', reducer });
const withSaga = injectSaga({ key: 'goalsList', saga });

export default compose(
  withReducer,
  withSaga,
  withConnect,
)(GoalsList);
