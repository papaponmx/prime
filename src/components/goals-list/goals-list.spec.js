import React from 'react';
import GoalsList from '.';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

Enzyme.configure({ adapter: new Adapter() });


describe('Goals list', () => {  
  const goalProps = {
    list: [
      {
        id: 1,
        name: 'Get swole',
      },
      {
        id: 2,
        name: 'Eat clean',
      },
      {
        id: 3,
        name: 'Win more',
      },
    ]
  }

  const component = shallow(
    <GoalsList goals={goalProps} />
  );
  const meThisWeek = component.find('li');

  it('Should render', () => {  
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('Should render a list of components', () => {  
    expect(meThisWeek.contains(<li>Win More</li>));
  });
});
