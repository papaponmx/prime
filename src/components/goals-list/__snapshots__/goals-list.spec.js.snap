// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Goals list Should render 1`] = `
<ul
  className="goals-list"
>
  <li
    key="1"
  >
    Get swole
  </li>
  <li
    key="2"
  >
    Eat clean
  </li>
  <li
    key="3"
  >
    Win more
  </li>
</ul>
`;
